lib/Models/Terria.js

```diff
@@ -550,17 +550,25 @@ function generateInitializationUrl(url) {
 
 function loadInitSources(terria, initSources) {
     return when.all(initSources.map(loadInitSource.bind(undefined, terria)), function(initSources) {
-        var promises = [];
+        var promises = [], updateSources = [];
 
         for (var i = 0; i < initSources.length; ++i) {
             var initSource = initSources[i];
             if (!defined(initSource)) {
                 continue;
             }
-            promises.push(terria.addInitSource(initSource));
+
+            if (initSource.catalogOnlyUpdatesExistingItems) {
```

Comment:
hmm there might be a deeper problem then (with the existing code I mean).  The intention is that the initSources would layer.  Later initSources are applied on top of earlier ones.  But that isn't necessarily true as soon as any part of applying an init source is asynchronous, right?

If we instead change the code to do `addInitSource` on the first init source, and wait for its promise to resolve before doing `addInitSource` on the second, I think that would eliminate the need for the two stages you've added here, and possibly also fix some other current/potential weird problems.  Does that sound right?